egs/wsj/s5/utils/lang/add_unigrams_arpa.pl

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env perl
+
+# Copyright 2018  Xiaohui Zhang
+# Apache 2.0.
```

Comment:
I notice there are no scripts in this commit that use this.  Is there any example that it wouldn't be too disruptive or time-consuming to check in?

Reply:
I've finished tuning the OOV recovery recipe (which used all scripts involved in this PR) for swbd-eval2000 & rt03. Basically, this recipe (unk-decoding + p2g + expanding lexicon & (RNN)LMs + 4-gram rescoring + RNNLM rescoring) results in much better CER on OOVs and matching overall WERs on eval2000/rt03 vs. a clean baseline without unk modeling (sorry, there's no improvement on overall WER, which makes sense since there are only hundreds of OOVs. Previously I mis-interpreted the improvements by pruned-RNNLM re-scoring as from my recipe.) The recipe is a bit ugly for now but you can skim it if you have time /export/b15/xzhang/swbd/s5c/oov_recovery.sh. I'll discuss more details with you on tmr's meeting. Besides this script, there are a few other changes needed to be committed: 1. modify lattice-arc-post so that the starting and ending state of each arc could be printed.  2. a script doing P2G based on sequitur is also needed.

Reply:
I'm planning to do more experiments on other datasets with more OOVs though.